internal/backend/auth/descope.go

```diff
@@ -0,0 +1,83 @@
+package auth
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/descope/go-sdk/descope/client"
+	"github.com/descope/go-sdk/descope/logger"
+	"go.uber.org/zap"
+)
+
+type descopeAuthenticator struct {
+	cfg  *Config
+	z    *zap.Logger
+	c    *client.DescopeClient
+	name string
+}
+
+func stringToLogLevel(level string) logger.LogLevel {
+	switch level {
+	case "none":
+		return logger.LogNone
+	case "debug":
+		return logger.LogDebugLevel
+	}
+
+	return logger.LogInfoLevel
+}
+
+func newDescopeAuthenticator(cfg *Config, z *zap.Logger) (Authenticator, error) {
+	if cfg.ProjectID == "" {
+		return nil, ErrInvalidAuthProviderConfiguration
+	}
+
+	ll := stringToLogLevel(cfg.LogLevel)
+
+	descopeClient, err := client.NewWithConfig(&client.Config{ProjectID: cfg.ProjectID, LogLevel: ll})
+	if err != nil {
+		z.Error("failed initiating descope: %w", zap.Error(err))
```

Comment:
Remove `": %w"`, the `zap.Error()` is a separate field in the log, not a part of the error message.

Reply:
good point, fixed